openmdao/components/interp_util/interp_algorithm.py

```diff
@@ -239,6 +258,282 @@ def interpolate(self, x, idx, slice_idx):
+            high = highbound
+
+        while x > grid[high]:
+            last_index = high
+            high += inc
+            if high >= highbound:
+
+                # Check if we're off of the top end
+                if x > grid[highbound]:
+                    return highbound, 1
+
+                high = highbound
+                break
+            inc += inc
+
+        # Bisection
+        while high - last_index > 1:
+            low = (high + last_index) // 2
+            if x < grid[low]:
+                high = low
+            else:
+                last_index = low
+
+        return last_index, 0
+
+    def evaluate(self, x, slice_idx=None):
+        """
+        Interpolate on this grid.
+
+        Parameters
+        ----------
+        x : ndarray
+            The coordinates to interpolate on this grid.
+        slice_idx : None
+            Only needed for API compatibility.
+
+        Returns
+        -------
+        ndarray
+            Interpolated values.
+        ndarray
+            Derivative of interpolated values with respect to independents.
+        ndarray
+            Derivative of interpolated values with respect to values.
+        ndarray
+            Derivative of interpolated values with respect to grid.
+        """
+        idx, _ = self.bracket(x)
+        result, d_dx, d_values, d_grid = self.interpolate(x, idx)
+
+        return result, d_dx, d_values, d_grid
+
+    def evaluate_vectorized(self, x, slice_idx=None):
+        """
+        Interpolate on this grid.
+
+        Parameters
+        ----------
+        x : ndarray
+            The coordinates to interpolate on this grid.
+        slice_idx : None
+            Only needed for API compatibility.
+
+        Returns
+        -------
+        ndarray
+            Interpolated values.
+        ndarray
+            Derivative of interpolated values with respect to independents.
+        ndarray
+            Derivative of interpolated values with respect to values.
+        ndarray
+            Derivative of interpolated values with respect to grid.
+        """
+        idx, _ = self.bracket(x)
+        result, d_dx, d_values, d_grid = self.interpolate_vectorized(x, idx)
+
+        return result, d_dx, d_values, d_grid
+
+    def interpolate(self, x, idx):
+        """
+        Compute the interpolated value.
+
+        This method must be defined by child classes.
+
+        Parameters
+        ----------
+        x : ndarray
+            The coordinates to interpolate on this grid.
+        idx : int
+            List of interval indices for x.
+
+        Returns
+        -------
+        ndarray
+            Interpolated values.
+        ndarray
+            Derivative of interpolated values with respect to independents.
+        ndarray
+            Derivative of interpolated values with respect to values.
+        ndarray
+            Derivative of interpolated values with respect to grid.
+        """
+        pass
```

Comment:
This should probably raise a NotImplementedError

Reply:
Replaced